src/main/java/edu/oregonstate/cs361/battleship/Hard.java

```diff
@@ -1,7 +1,354 @@
 package edu.oregonstate.cs361.battleship;
 
+import java.util.Random;
+
 /**
  * Created by Keana on 3/15/2017.
  */
-public class Hard extends Mode {
+public class Hard extends Mode{
+
+
+        public Hard(){
+
+        }
+        public void placeShips(Military computer_aircraftCarrier,  Military computer_battleship,  Military computer_submarine, Civilian computer_Dinghy,  Civilian computer_Clipper){
+
+
+                int max = 10;
+                int min = 1;
+                Random random = new Random();
+                boolean validLocation = false;
+
+                //Aircraft Carrier
+
+                while(validLocation == false) {
+
+
+                        int randDir = random.nextInt(2);
+
+                        int randRow = random.nextInt(max - min + 1) + min;
+                        int randCol = random.nextInt(max - min + 1) + min;
+
+                        Coordinate coor = new Coordinate(randRow, randCol);
+
+                        computer_aircraftCarrier.setStart(coor);
+                        if (randDir == 0) {
+                                int endRow = randRow + 4;

```

Comment:
This section of code is repeated quite frequently throughout this section. Have you considered breaking this into a helper function?